File: benchmarks/pandas/bench_str_join.py
"""Benchmark: str_join — str.join on 100k list-of-strings Series values"""
import json, time
import pandas as pd

ROWS = 100_000
WARMUP = 3
ITERATIONS = 10

data = [[f"a{i % 10}", f"b{i % 5}", f"c{i % 3}"] for i in range(ROWS)]
s = pd.Series(data)

for _ in range(WARMUP):
    s.str.join("-")

start = time.perf_counter()
for _ in range(ITERATIONS):
    s.str.join("-")
total = (time.perf_counter() - start) * 1000

print(json.dumps({
    "function": "str_join",
    "mean_ms": total / ITERATIONS,
    "iterations": ITERATIONS,
    "total_ms": total,
}))

File: benchmarks/tsb/bench_str_join.ts
/**
 * Benchmark: str_join — str.join on 100k list-of-strings Series values
 */
import { Series } from "../../src/index.js";

const ROWS = 100_000;
const WARMUP = 3;
const ITERATIONS = 10;

// Series where each element is a list of strings (already split)
const data = Array.from({ length: ROWS }, (_, i) => [`a${i % 10}`, `b${i % 5}`, `c${i % 3}`]);
const s = new Series({ data });

for (let i = 0; i < WARMUP; i++) {
  s.str.join("-");
}

const start = performance.now();
for (let i = 0; i < ITERATIONS; i++) {
  s.str.join("-");
}
const total = performance.now() - start;

console.log(
  JSON.stringify({
    function: "str_join",
    mean_ms: total / ITERATIONS,
    iterations: ITERATIONS,
    total_ms: total,
  }),
);
